pkg/pgclient/client_test.go

```diff
@@ -183,3 +184,57 @@ func TestHealthCheck(t *testing.T) {
 		t.Fatal("health check method not called when expected")
 	}
 }
+
+func TestNewClient(t *testing.T) {
```

Comment:
Could use a bit more test cases, to at least cover `GetNumConnections` and `ParseConfig` errors.

Reply:
In `CreateNewClient` we are creating` connectionPool` adding more test cases to validate ParseFlags config is failing the func call as DB calls will fail. Any thoughts/workaround to handle this?

Reply:
I just want to make sure we covered all the cases logically using the unit test. If you feel we achieved that, you can leave the test as is.